cmd/tailscale/cli/up.go

```diff
@@ -546,6 +580,20 @@ func runUp(ctx context.Context, args []string) error {
 	}
 }
 
+func printUpDoneJSON(state, errorString string) {
+	type upDoneJSON struct {
```

Comment:
let's merge these two types together... they're all one thing the client will parse. I know you're not using Go, but imagine you were using Go... you'd want one type to unmarshal into.

Then use omitempty annotations on the fields that aren't always present including BackendState.

Reply:
Done.